http/middleware/throttle.go

```diff
@@ -1,63 +1,51 @@
 package middleware
 
 import (
-	"crypto/md5"
-	"encoding/hex"
 	"fmt"
 	"strconv"
 	"time"
 
 	httpcontract "github.com/goravel/framework/contracts/http"
 	"github.com/goravel/framework/http"
 	httplimit "github.com/goravel/framework/http/limit"
-	"github.com/goravel/framework/support/carbon"
+)
+
+const (
+	// HeaderRateLimitLimit, HeaderRateLimitRemaining, and HeaderRateLimitReset
+	// are the recommended return header values from IETF on rate limiting. Reset
+	// is in UTC time.
+	HeaderRateLimitLimit     = "X-RateLimit-Limit"
+	HeaderRateLimitRemaining = "X-RateLimit-Remaining"
+	HeaderRateLimitReset     = "X-RateLimit-Reset"
+
+	// HeaderRetryAfter is the header used to indicate when a client should retry
+	// requests (when the rate limit expires), in UTC time.
+	HeaderRetryAfter = "Retry-After"
 )
 
 func Throttle(name string) httpcontract.Middleware {
 	return func(ctx httpcontract.Context) {
 		if limiter := http.RateLimiterFacade.Limiter(name); limiter != nil {
 			if limits := limiter(ctx); len(limits) > 0 {
 				for _, limit := range limits {
-					if instance, ok := limit.(*httplimit.Limit); ok {
-						key, timer := key(ctx, name, instance)
-						currentTimes := 1
-
-						if http.CacheFacade.Has(timer) {
-							value := http.CacheFacade.GetInt(key, 0)
-							if value >= instance.MaxAttempts {
-								expireSecond := http.CacheFacade.GetInt(timer, 0) + instance.DecayMinutes*60
-								ctx.Response().Header("X-RateLimit-Reset", strconv.Itoa(expireSecond))
-								ctx.Response().Header("Retry-After", strconv.Itoa(expireSecond-int(carbon.Now().Timestamp())))
-								if instance.ResponseCallback != nil {
-									instance.ResponseCallback(ctx)
-									return
-								} else {
-									ctx.Request().AbortWithStatus(httpcontract.StatusTooManyRequests)
-									return
-								}
-							} else {
-								var err error
-								if currentTimes, err = http.CacheFacade.Increment(key); err != nil {
-									panic(err)
-								}
-							}
-						} else {
-							expireMinute := time.Duration(instance.DecayMinutes) * time.Minute
+					if instance, exist := limit.(*httplimit.Limit); exist {
+						tokens, remaining, reset, ok, err := instance.Store.Take(ctx, key(ctx, instance))
+						if err != nil {
+							response(ctx, instance)
+							return
```

Comment:
What's the `err`?

Reply:
> What's the `err`?

I checked the file https://github.com/sethvargo/go-limiter/blob/main/memorystore/store.go , all return is `nil`, so this can be removed

Reply:
> Not nil:
> 
> <img alt="image" width="884" src="https://private-user-images.githubusercontent.com/24771476/342784478-ffff8b74-a06d-4d7d-8110-aa301267093e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzMzM2MDQsIm5iZiI6MTcxOTMzMzMwNCwicGF0aCI6Ii8yNDc3MTQ3Ni8zNDI3ODQ0NzgtZmZmZjhiNzQtYTA2ZC00ZDdkLTgxMTAtYWEzMDEyNjcwOTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI1VDE2MzUwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlNThkOTUxNzg5MWQ5YjE3YzMzYTFjYmMzMWY0NzMzNGY5MTQyYTExMzdjYjc4MDc4OGY2ZWE5MWFkZGY2YTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.B1wLwR9pJWS_vVCyc5QzwRd5s58JxozHg183DjnNFJA">

Seems no problem, because we doesn't need to `Close` the limiter.